cmd/estuary-shuttle/rpc.go

```diff
@@ -278,18 +280,17 @@ func (d *Shuttle) handleRpcTakeContent(ctx context.Context, cmd *drpc.TakeConten
 		if err != nil {
 			return err
 		}
+
 		if count > 0 {
-			if count > 1 {
-				log.Errorf("have multiple pins for same content: %d", c.ID)
-			}
 			continue
 		}
 
-		if err := d.addPin(ctx, c.ID, c.Cid, c.UserID, true); err != nil {
-			return err
-		}
+		go func(c drpc.ContentFetch) {
```

Comment:
why did we need to create a go routine here?

Reply:
to make sure each pin request does not block each other, even if 1 errors - best effort